content/docs/learn/typed-errors/from-either-to-raise.md

```diff
@@ -0,0 +1,181 @@
+    }
+  }
+```
+
+The translation into the `Raise` DSL describes the same sequence
+of steps, but using a more sequential style:
+
+```kotlin
+fun tururu(n: Int): Either<Error, String> = either {
+  val s = f(n).bind()
+  val t = g(s).bind()
+  t.summarize()
+}
+```
+
+There are two important steps in this translation. At the beginning of
+the function, we have the `either` builder, which (1) indicates that
+the result of the function is `Either` of some error type and (2)
+creates a new scope in which you can use the `Raise` DSL.
+
+The second step is that every time that you have an `Either` value
+resulting from another computation (like `f` and `g` above), you need
+to call `bind`. This "injects" any potential error value in the context,
+and continues normal execution otherwise.
+In this style you don't need to use any function to compose computations,
+you use regular Kotlin idioms with some `bind`s sprinkled. In fact
+you could have written the previous code in one single line:
+
+```kotlin
+fun tururu(n: Int): Either<Error, String> = either {
+  g(f(n).bind()).bind().summarize()
+}
+```
+
+How you split your code into local values no longer depends on the
+structure of your functions, as is the case with `flatMap`, but rather
+on the logical decomposition you want in your code.
+
+Arrow provides different builders for different return types
+(`either`, `option`, `result`), but regardless of the one you choose
+you always use `bind` for the injection phase.
+
+## Returning with logical error
+
+Using wrapper types you often have a specific constructor that represents
+the error case. For example, `Left` signals failure when using `Either`.
+Using the `Raise` DSL you no longer need to remember each of them, 
+failure is always signaled using `raise`.
+
+```kotlin
+fun tururuThatRaise(n: Int): Either<Error, String> = either {
+  if (n < 0) raise(Error.NegativeInput)
+  val s = f(n).bind()
+  val t = g(s).bind()
+  t.summarize()
+}
+```
+
+Calling `raise` immediately ends the execution of the current block. In the example
+above, calling `h(-1)` ends up in `raise`, which in the case of an `either` block
+means resulting in `Either.Left(Error.NegativeInput)`.
+
+:::info Somebody said "exceptions"?
+
+The propagation and early return of the `Raise` scope look similar to exceptions.
+However, typed errors should be used for _logical_ errors — problems that have
+a place in your domain model — as opposed to _exceptional_ cases — which represent
+circumstances that are difficult to recover from.
+
+:::
+
+## Transforming the error values
+
+The fact that both `f` and `g` share the same `Error` type is key for
+the previous code blocks to compile. In every `Raise` scope, we have one _single_
+error type which you can `raise` or `bind` in the whole block of code.
+If this is not the case, you have to _bridge_ the different error types, which
+is done using functions like `mapLeft`.
+
+```kotlin
+fun tarara(n: Int): Either<Error, String> =
```

Comment:
What is with these example function names?  It's actually distracting trying to figure out what they are referencing, since they're not the industry standard nonsense words.  (There's even an RFC for that.)